packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1201,31 +1170,26 @@ export class ContainerRuntime
 	public get idCompressorMode() {
 		return this.sessionSchema.idCompressorMode;
 	}
+
 	/**
 	 * See IContainerRuntimeBase.idCompressor() for details.
 	 */
 	public get idCompressor() {
-		// Expose ID Compressor only if it's On from the start.
-		// If container uses delayed mode, then we can only expose generateDocumentUniqueId() and nothing else.
-		// That's because any other usage will require immidiate loading of ID Compressor in next sessions in order
-		// to reason over such things as session ID space.
-		if (this.idCompressorMode === "on") {
-			assert(this._idCompressor !== undefined, 0x8ea /* compressor should have been loaded */);
-			return this._idCompressor;
+		// During a schema migration, the idCompressor is eagerly loaded but should not be exposed until
+		// the document schema change is committed.
+		if (this.idCompressorMode !== "on") {
+			return undefined;
 		}
-	}
 
-	/**
-	 * True if we have ID compressor loading in-flight (async operation). Useful only for
-	 * this.idCompressorMode === "delayed" mode
-	 */
-	protected _loadIdCompressor: Promise<void> | undefined;
+		assert(this._idCompressor !== undefined, 0x8ea /* compressor should have been loaded */);
+		return this._idCompressor;
+	}
 
 	/**
 	 * See IContainerRuntimeBase.generateDocumentUniqueId() for details.
 	 */
 	public generateDocumentUniqueId() {
-		return this._idCompressor?.generateDocumentUniqueId() ?? uuid();
+		return this.idCompressor?.generateDocumentUniqueId() ?? uuid();
```

Comment:
Most references to the '_idCompressor' field have been changed to use the 'idCompressor' property.

The difference being that the 'idCompressor' property continues to return 'undefined' until the migration op transitions the IdCompressorMode to "on".